test/util_test.dart

```diff
@@ -155,6 +158,74 @@ void overReactExample() {}''';
         expect(stripPrivateGeneratedPrefix(r'Foo_$Bar'), r'Foo_$Bar');
       });
     });
+
+    group('hasComment()', () {
+      final content = '''
+          main() { // Comment0
+            // Comment1
+            // Comment2
+            var one;
+            var two;
+            
+            // Comment3
+          }
+        ''';
+
+      final sourceFile = SourceFile.fromString(content);
+      final astNode = parseCompilationUnit(content);
+
+      test('correctly finds a comment', () {
+        expect(hasComment(astNode, sourceFile, 'Comment0'), isTrue);
+      });
+
+      test('breaks after finding first comment', () {
+        expect(hasComment(astNode, sourceFile, 'Comment1'), isFalse);
+      });
+
+      test('only finds a comment within one line of the provided node', () {
+        expect(hasComment(astNode, sourceFile, 'Comment2'), isFalse);
+      });
+    });
+
+    group('allComments()', () {
+      final content = '''
+          main() {  
+            // Comment1
+            // Comment2
+            var one;
+            var two;
+            
+            // Comment3
+          }
+        ''';
+
+      final astNode = parseCompilationUnit(content);
+      int commentCount;
+
+      setUp(() {
+        commentCount = 0;
+      });
+
+      test('correctly iterates over all comments', () {
+        commentCount = allComments(astNode.beginToken).length;
+
+        expect(commentCount, equals(3));
+      });
+
+      test('only finds comments after the provided starting node', () {
+        Token begin = astNode.beginToken;
+        Token firstVar = begin;
+
+        // Traverse the tokens until we get to var one.
+        while (firstVar.value() != 'one' && !firstVar.isEof) {
+          firstVar = firstVar.next;
+        }
+
+        commentCount = allComments(firstVar).length;
+
+        expect(commentCount, equals(1));
```

Comment:
#supernit `equals` is the default matcher, so its superfluous.

```suggestion
        expect(commentCount, 1);
```